From 3ac0062a27a34d6606a1275d979333bf8c09bed7 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Thu, 23 May 2019 14:40:30 +0300
Subject: [PATCH] [SW-0] pcie: update wait for link

Just like with Goya do not access PCIe blocks while perst is asserted
(it may jam bus)

Change-Id: I9808174c24164bd5aa37bfe03193fd938aabf379
---

diff --git a/gaudi/common.c b/gaudi/common.c
index d3b289d..8205883 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -16,6 +16,10 @@
 
 void preboot_pcie_wait_for_link(void)
 {
+	/* Wait for PRSTn */
+	while (!regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN))
+		;
+
 	while (1) {
 		unsigned int smlh, rdlh;
 
